### PR TITLE
Use ingredient_config resource to update the configuration.

### PR DESCRIPTION
**Pull request by sersut**
@jtimberman single kitchen run is looking good. Running the full suite now. 

Are we cool with changing the configuration from `Hash` to `String`? I am not sure how many people are using this but this will be a breaking change. We can still preserve the old functionality by moving the logic in the template to a helper method, deprecating the `configuration` parameter and introducing a new parameter called something other than `configuration` if we want to.

### CODE REVIEW

**Comment by jtimberman** [comment]
So, during standup we discussed having a compatibility release.

On looking into refactoring this to support that, I propose that we *not* have a compatibility release. We should just make a major version change here, but not merge / release w/o announcing to the community and giving it a few days. :cake:

File: attributes/default.rb [hunk 1]
```diff
@@ -36,18 +36,18 @@
...
-#           }
-#         }
+#         'configuration' => <<-EOS
+#     nginx['ssl_port'] = 4433#
+#     EOS
 #       }
 #     )
 #
-default['chef-server']['configuration'] = {}
+default['chef-server']['configuration'] = ""
```
Comment (jtimberman) [inline attributes/default.rb]: Rubocop says `''` :smile_cat:

File: recipes/default.rb [hunk 2]
```diff
@@ -27,27 +27,18 @@
 chef_ingredient 'chef-server' do
   version node['chef-server']['version']
   package_source node['chef-server']['package_source']
+  config <<-EOS
+topology "standalone"
```
Comment (jtimberman) [inline recipes/default.rb]: Perhaps we should make a `node['chef-server']['topology']` attribute, and default to `standalone`? That would give us some flexibility for reuse.

File: attributes/default.rb (hunk 1 above)
Comment (sersut) [inline attributes/default.rb]: :smile:

File: README.md [hunk 3]
```diff
@@ -150,19 +139,48 @@ Then pass this file to the initial chef-solo command:
...
+nginx['cache_max_size'] = '3500m'
+EOS
+```
+
+In a `dna.json` file, we need to insert a `\n` newline character.
+
+```json
+{
+  "chef-server": {
+    "configuration': "notification_email 'chef-server@example.com'\nnginx['cache_max_size'] = '3500m'"
```
Comment (jtimberman) [inline README.md]: wtf?

**Pull request merged**